### PR TITLE
PLT-171:Added unit test for update and delete for cluster group

### PR DESCRIPTION
**Pull request by SivaanandM**

### CODE REVIEW

File: pkg/client/client.go [hunk 1]
```diff
@@ -54,6 +54,8 @@ type V1Client struct {
 	CreateClusterGroupFn func(*models.V1ClusterGroupEntity) (string, error)
 	GetClusterGroupFn    func(string) (*models.V1ClusterGroup, error)
 	GetApplicationFn     func(string) (*models.V1AppDeployment, error)
+	UpdateClusterGroupFn func(string, *models.V1ClusterGroupHostClusterEntity) error
```
Comment (nikchern) [inline pkg/client/client.go:57]: Could you please group these functional objects together by Object type i.e. application is separated from cluster group for example.

**Review by nikchern — CHANGES_REQUESTED** [review]

File: spectrocloud/resource_cluster_group.go [hunk 2]
```diff
@@ -221,9 +221,14 @@ func flattenClusterGroup(clusterGroup *models.V1ClusterGroup, d *schema.Resource
 
 func resourceClusterGroupUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	c := m.(*client.V1Client)
-
 	// Warning or errors can be collected in a slice type
 	var diags diag.Diagnostics
+	// Unit test handler
+	if c.UpdateClusterGroupFn != nil {
```
[the remark this answers is not in the document]
Comment (SivaanandM) [reply spectrocloud/resource_cluster_group.go:227]: I tried but could not change the tf state from a unit test.

File: pkg/client/client.go (hunk 1 above)
Comment (SivaanandM) [reply pkg/client/client.go:57]: updated

File: spectrocloud/resource_cluster_group.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (SivaanandM) [reply spectrocloud/resource_cluster_group.go:227]: As discussed, for now, this is a workaround. will d more research on the same.

**Review by nikchern — APPROVED** [review]

**Pull request merged**